### PR TITLE
Update our code base, tests and endpoints, to use our new permissions model.

### LINKED ISSUE #241: [5] Update tests to reflect new sitewide permissions
This will mostly involve auditing existing tests to ensure they work with the new permissions, but some new tests may need to be written.

### PR DESCRIPTION
**Pull request by LyonesGamer**
Fixes #241 and #242.

### CODE REVIEW

File: tests/times.js [hunk 1]
```diff
@@ -100,6 +100,226 @@ module.exports = function(expect, request, baseUrl) {
...
+        });
+      });
+    });
+
+    it('returns all times in the database to a sitewide spectator',
+    function(done) {
+      const oldUser = user;
+      const oldPass = password;
+
+      user = 'patcht';
```
Comment (athai) [inline tests/times.js]: Wouldn't it be better to use 'mrsj' for the spectator tests? Considering he only has site_spectator permission whereas 'patcht' has both site_manager and site_spectator perms.

File: tests/times.js [hunk 2]
```diff
@@ -100,6 +100,226 @@ module.exports = function(expect, request, baseUrl) {
...
+          done();
+        });
+      });
+    });
+
+    it("returns a project spectator's set of times", function(done) {
+      const oldUser = user;
+      const oldPass = password;
+
+      user = 'deanj';
```
Comment (athai) [inline tests/times.js]: 'deanj' doesn't have spectator permissions for gwm. You did add him as a project member though.

**Comment by ElijahCaine** [comment]
Nothing jumps out at me as being a red flag, but another pair of eyes would be good. +1

**Comment by athai** [comment]
Looks good to me. +1

**Pull request merged**